### ISSUE TITLE
Salt Level not working after Boost command is sent

### ISSUE THREAD

**Issue by art0** [DutchForce75/pyZodiacTri#2]
192.168.1.224 - - [10/Sep/2019 19:14:26] "GET /setChlorine?pct=101 HTTP/1.1" 200 -
RS485           : <-- 1002 50 00  62 1003 
RS485           : -->  
RS485           : <-- 1002 50 11 65 d8 1003 
RS485           : -->  
RS485           : <-- 1002 50 00  62 1003 
RS485           : --> 1002 00 01 0000 13 1003 
RS485           : <-- 1002 50 11 65 d8 1003 
RS485           : --> 1002 00 12 0001 25 1003 
RS485           : <-- 1002 50 00  62 1003 
RS485           : --> 1002 00 01 0000 13 1003 
RS485           : <-- 1002 50 11 65 d8 1003 
RS485           : --> 1002 00 12 0001 25 1003 
RS485           : <-- 1002 50 00  62 1003 
RS485           : --> 1002 00 01 0000 13 1003 
RS485           : <-- 1002 50 11 65 d8 1003 
RS485           : --> 1002 00 12 0001 25 1003 
RS485           : <-- 1002 50 00  62 1003 
RS485           : --> 1002 00 01 0000 13 1003 
RS485           : <-- 1002 50 11 65 d8 1003 
RS485           : --> 1002 00 12 0001 25 1003 

Salt
Add Salt

Displays "0"

**Comment by DutchForce75** [comment]
Seems that your the software version you are using is responding differently to setting the chlorine percentage:
RS485 : <-- 1002 50 11 65 d8 1003
RS485 : --> 1002 00 12 0001 25 1003
Fourth byte is in your case 0x12, and in my case 0x16 and seems to be the software version.

What is your Ext Controller setting atm?

**Comment by art0** [comment]
Sounds like I needed to restart the box, it was version Rev K

RS485           : <-- 1002 50 11 65 d8 1003 
RS485           : --> 1002 00 16 28000000 50 1003 

But if set up a % of Chlore, the only thing i see in the display is

PH/ACL CONTROL and not the OUTPUT thingy.
So i figured out that first i needed to set up the "ACL Override" in order to see the OUTPUT:  ##%

Do you know  when using the % of SWG, how often the time is sampled? I am pretty sure the cell is always at 100% power.

**Comment by DutchForce75** [comment]
Don't know how often it samples, you could set it to 0%, from memory I think it shuts off?
Like you suggested, I think it its time switched instead of voltage regulated

Great that it seems to work on your side too btw!

**Comment by art0** [comment]
I still debugging a race condition, not sure the "toggle" thingy is really providing the correct feedback. Any chance to log also the RS485 messages to a single log with the webserver requests?

ORP is far from perfect.

ACL 1: 500 mV
ACL 2: 550 mV
ACL 3: 600 mV
ACL 4: 650 mV
ACL 5: 700 mV
ACL 6: 750 mV

In the presence of CYA over 50 ppm, the ORP drops to 300-350 mV when the water is 25-26C. So the SWG is pretty much useless.

So I am trying to create my own control and just look at how green is the water with a webcam. :) Next phase is to read the RGB of the water colour

Let me know where to send you a beer :)

**Comment by DutchForce75** [comment]
Not sure what works on your end and what not.

* Do you have the webserver running?
* Have you enabled the Talk To Zodiac radiobutton? This one needs to be set on. Only then will packets be sent to the Tri.
* Have you tried to set the Chloric slider to it's max? This is the Boost function and the led should be turned on on the Tri. 
* Do you see the connection icon change in the Tri's display when you send the packages to the Tri?

**Comment by DutchForce75** [comment]
> Any chance to log also the RS485 messages to a single log with the webserver requests?

You could try to fire the webserver like this, can't test it right now
python2 /yourdir/webserver.py > log.txt

**Comment by DutchForce75** [comment]
To be more clear: the Talk To Zodiac radiobutton on the webpage needs to be switched on for as long as you want the Zodiac Tri to be on those settings. It will keep connecting to the Zodiac for as long as it's switched On. As soon as you turn it to Off, it won't send any messages to the Tri, and the Tri will switch to Off mode.

**Comment by art0** [comment]
* logging: 
  I tried to forward the stderr to the log 2>&1 and still do not get a good log. Maybe worth to log also the time in RS comms to have one single log 

Radiobutton
* I guess you are constantly probing the SWG to keep it alive. I realized that when running multiple "tabs" in the browser, there is no way to know the status of the connection. Maybe is worth to save that in a webserver variable and force a refresh?

**Comment by DutchForce75** [comment]
Yes there is a lot to be done to make this a proper software release!

Only way to refresh right now is hitting refresh in the webbrowser. If Link is displaying "connected" then the webserver is still receiving replies and the Tri should be up and running. Otherwise the webserver isn't getting messages from the Tri and connection is lost or hasnt started at all

For a future release it would be nice to have some sort of Ajax script pushing the latest status to the web client, but for my situation I'd like it to be as simple as possible since the internet connection on the Zodiac's side is not that good. 

For now I can at least shut off the Tri to chlore shock the pool with another pump. And I can see if the filterpump is running. Next up is reading ORP and pH values

**Comment by art0** [comment]
off the record, what pump you use to "chlore shock" and how do you control the volume? I am using the well known technique of "dropping supermarket bleach" :)

**Comment by DutchForce75** [comment]
I use this one: https://www.toppy.nl/product/8183/520/zwembad/desinfectietechniek/doseertechniek/blue-lagoon-peristaltische-doseerpomp-met-vaste-flow-3ltr-uur.html

**Comment by art0** [comment]
How do you inject the chlore into the pool?

The "Pod" that comes with the Zodiac includes a PH- pump, not sure if i should mix the Chlore and the H2SO4 in the same inlet.

**Comment by DutchForce75** [comment]
Yes it's injecting that into that Pod, and that's why I need to be able to shut off the Zodiac when I chlore shock, otherwise the Zodiac will instantly add pH- too which is not desirable.

Tbh Im fairly new to pool maintenance so I am still learning.
>

**Comment by art0** [comment]
So you have made a T-junction in the tube that goes into the pod? Picture :)

**Comment by DutchForce75** [comment]
![IMG_7391](https://user-images.githubusercontent.com/55048602/64695724-15128500-d49d-11e9-93c5-40444ffc18bb.PNG)

**Comment by art0** [comment]
Has the pod 2 places to inject?

ORP
PH
Flow
PH-
Chlore?

**Comment by DutchForce75** [comment]
Yes I added another inject valve on the pod (already there) Not the flow switch, that's the other thing on the right

**Comment by art0** [comment]
I am not sure i understand. 
-Do you use a 4-ways POD and removed the Flow sensor?
- In mine i have PH-, ORP, PH, Flow
- You run the flow sensor somewhere else?

Where did you find the adapter?

**Comment by art0** [comment]
Now i understand. In my setup the "flow switch" is connected to the  "Pod" so i do not have an extra place to inject the Chlore.

**Comment by DutchForce75** [comment]
Im considering to place the injection point somewhere further away from the pod in the direction of the pool. It might prevent switching off the Zodiac when shocking.
